File: MyTodoApp_Backend/src/main/java/com/example/MyTodo/Service/TaskService.java
package com.example.MyTodo.Service;

import com.example.MyTodo.DTO.Request.CreateTaskRequest;
import com.example.MyTodo.DTO.Response.TaskResponse;

import java.util.List;

public interface TaskService {

    List<TaskResponse> getAllTasks();

    TaskResponse getTaskById(Long id);

    TaskResponse createTask(CreateTaskRequest request);

    TaskResponse updateTask(Long id, CreateTaskRequest request);

    void deleteTask(Long id);

    TaskResponse updateStatus(Long id, boolean completed);

    List<TaskResponse> searchByTitle(String keyword);

    List<TaskResponse> filterByStatus(boolean completed);

}

File: MyTodoApp_Frontend/src/services/TaskService.js
import axios from "axios";

const API_URL = "http://localhost:8080/tasks";

const taskService = {

    getAllTasks() {
        return axios.get(API_URL);
    },

    getTaskById(id) {
        return axios.get(`${API_URL}/${id}`);
    },

    createTask(task) {
        return axios.post(API_URL, task);
    },

    updateTask(id, task) {
        return axios.put(`${API_URL}/${id}`, task);
    },

    deleteTask(id) {
        return axios.delete(`${API_URL}/${id}`);
    },

    updateStatus(id, completed) {
        return axios.patch(`${API_URL}/${id}/status`, {
            completed
        });
    },

    searchTask(keyword) {
        return axios.get(`${API_URL}/search?keyword=${keyword}`);
    }

};

export default taskService;